### PR TITLE
implement web share target

### LINKED ISSUE #5850: Implement Web Share Target
The [Web Share Target API](https://github.com/WICG/web-share-target/blob/master/docs/explainer.md) would allow native apps (e.g. on Android) to share text (and possibly other media) with Mastodon.

Use case:
 - You find a nice quote you want to post on Mastodon
 - You select it and click "Share"
 - You select Mastodon from the share options (it will show up like a native app)
 - Mastodon opens and the text is pasted into the compose field.

This is currently supported in Chrome Canary on Android ([source](https://twitter.com/kennethrohde/status/935762149116465154)) - but you can feature detect it such that it will automatically become available to everyone as soon as the feature lands in Chrome Stable (and other browsers).

### PR DESCRIPTION
**Pull request by rinsuki**
video: https://mstdn.maud.io/@rinsuki/99363572785336264
close #5850

### CODE REVIEW

File: app/controllers/shares_controller.rb [hunk 1]
```diff
@@ -14,13 +14,16 @@ def show
   private
 
   def initial_state_params
+    text = params[:text] || ''
+    text = params[:title] + ' ' + text if params[:title]
+    text += ' ' + params[:url] if params[:url]
```
Comment (ykzts) [inline app/controllers/shares_controller.rb]:
```ruby
text = params.slice(:title, :text, :url).values.compact.join(' ')
```

or

```ruby
text = [params[:title], params[:text], params[:url]].compact.join(' ')
```

**Comment by ykzts** [comment]
cc @sorin-davidoi

**Review by Gargron — APPROVED** [review]

**Comment by jessehattabaugh** [comment]
Has this been confirmed working on any devices? I am on Android 7 Chrome 68/Canary 71 and Mastodon doesn't even appear in the "share via" picker. I've gotten my own app to appear in the picker, but clicking it does nothing. Not a Mastodon bug, but I'm just trying to decide if I need to open a ticket with Chrome

**Comment by Gargron** [comment]
I've seen it work in screenshots of Chrome but as I no longer use an Android device I can't help you further. Maybe @sorin-davidoi

**Comment by sorin-davidoi** [comment]
Haven't tested it recently I'm afraid.

**Comment by rinsuki** [comment]
How did you add Mastodon web-app? @jessehattabaugh 
I think, web share target is only working in added apk by Chrome "Add to Home Screen" banner.

**Comment by jessehattabaugh** [comment]
I "added to Home screen" on Android 7, Chrome 68 (with experimental flag
enabled) and Canary 71. chrome://webapks shows it is installed. The
official WST test page doesn't work either. The only place I've ever seen
it work is Chrome 69 Win10 w/ flag enabled.

**Pull request merged**